python/ray/tests/test_job.py

```diff
@@ -183,22 +183,28 @@ def test_job_timestamps(ray_start_regular):
 from time import sleep
 
 ray.init(address="{}")
+open("{}", "w+").close()
 
 print("My job id: ", str(ray.get_runtime_context().job_id))
 
 {}
 ray.shutdown()
     """
+    tmpfiles = ["/tmp/first.tmp", "/tmp/second.tmp"]
```

Comment:
Let's generate unique filenames here (e.g. with NamedTemporaryFile or mkstemp) to make sure that we avoid race conditions if multiple tests are running in parallel and accessing/checking the same files :)

Reply:
That was my first choice (i.e., using `NamedTemporaryFile`) but it gives permission errors. So, if the concern is race conditions then I can do a workaround by providing a fixed `dir`. Let me see what I can do.

Reply:
So, I noticed that when we use default `NamedTemporaryFile`, we get, 

```python
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
OSError: [Errno 22] Invalid argument: 'C:\\CONDA_~1\tmpif7o9w1j.tmp'
```

The reason is the name generated by `NamedTemporaryFile` has, `\tmp` as the prefix. `\t` gets treated as a tab instead of the component of a path in Windows. This is more of a Python bug as it forgot to put "\\tmp" instead of `\tmp` in `'C:\\CONDA_~1\tmpif7o9w1j.tmp'`. I pass the `prefix="_"` in `NamedTemporaryFile` to avoid any kind of escape sequences in the path string generated. This seems to work consistently on my machine. Hopefully it should work on CI as well.